### PR TITLE
Add QOI reader/writer

### PR DESCRIPTION
**Pull request by KristofferC**
Ref https://github.com/JuliaIO/FileIO.jl/pull/354#discussion_r776882059

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -17,8 +19,9 @@ ImageCore = "0.8.1, 0.9"
 Netpbm = "1.0"
 OpenEXR = "0.3"
 PNGFiles = "0.3"
+QOI = "1"
 TiffImages = "0.3, 0.4, 0.5"
-julia = "1.3"
+julia = "1.6"
```
Comment (KristofferC) [inline Project.toml:23]: Note

File: Project.toml [hunk 2]
```diff
@@ -5,9 +5,11 @@ version = "0.5.9"
 
 [deps]
 FileIO = "5789e2e9-d7fb-5bc7-8068-2c6fae9b9549"
+ImageCore = "a09fc81d-aa75-5fe9-8630-4744c3626534"
```
Comment (johnnychen94) [inline Project.toml]: This seems unnecessary to me.

```suggestion
```

**Review by johnnychen94 — APPROVED** [review]
Thanks for the typo fixes!

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -8,7 +8,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        julia-version: ['1.3', '1', 'nightly']
+        julia-version: ['1', 'nightly']
```
Comment (timholy) [inline .github/workflows/test.yml:11]: We should add 1.6 because 1 means 1.7, yet Project.toml allows 1.6.

Comment (johnnychen94) [reply .github/workflows/test.yml:11]: yeah fixed in #44

**Pull request merged**